core/document-types/page.php

```diff
@@ -64,4 +71,76 @@ public static function get_title() {
 	public static function get_plural_title() {
 		return esc_html__( 'Pages', 'elementor' );
 	}
+
+	public static function get_site_settings_url_config( $returnString ) {
+		$existing_elementor_page = self::get_elementor_page();
+		$site_settings_url = ! empty( $existing_elementor_page )
+			? self::get_elementor_edit_url( $existing_elementor_page->ID )
+			: self::get_elementor_create_new_page_url( $returnString ? self::SITE_IDENTITY_TAB : null);
+
+		if ($returnString) {
+			return $site_settings_url;
+		}
+
+		return [
+			'new_page' => empty( $existing_elementor_page ),
+			'url' => $site_settings_url,
+			'type' => static::URL_TYPE,
+		];
+	}
+
+	public static function get_elementor_create_new_page_url( $active_tab = null ): string {
+		$active_kit_id = Plugin::$instance->kits_manager->get_active_id();
+
+		if ( empty( $active_kit_id ) ) {
+			return Plugin::$instance->documents->get_create_new_post_url( 'page' );
+		}
+
+		$args = [
+			'active-document' => Plugin::$instance->kits_manager->get_active_id(),
+			'active-tab' => $active_tab,
+		];
+
+		return add_query_arg( $args, Plugin::$instance->documents->get_create_new_post_url( 'page' ) );
+	}
+
+	private static function get_elementor_edit_url( int $post_id ): string {
+		$active_kit_id = Plugin::$instance->kits_manager->get_active_id();
+		$document = Plugin::$instance->documents->get( $post_id );
+
+		if ( ! $document ) {
+			return '';
+		}
+
+		return add_query_arg( [ 'active-document' => $active_kit_id ], $document->get_edit_url() );
+	}
+
+	private static function get_elementor_page() {
+		$args = [
+			'meta_key' => Document::BUILT_WITH_ELEMENTOR_META_KEY,
+			'sort_order' => 'asc',
+			'sort_column' => 'post_date',
+		];
+		$pages = get_pages( $args );
```

Comment:
Could there maybe be an arg property to limit the amount of queried objects? That way you will save time connecting to DB and avoid the loop. WDYT?

Reply:
If you know a method like this, i am open for suggestions, i couldnt find any.

Reply:
https://developer.wordpress.org/reference/functions/get_pages/